specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,328 @@
+                auto session = m_devToolsSessionMap.find(sessionId);
+                if (session != m_devToolsSessionMap.end())
+                {
+                    m_devToolsTargetDescriptionMap.erase(session->second);
+                    m_devToolsSessionMap.erase(session);
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_targetDetachedToken));
+    receiver.reset();
+    CHECK_FAILURE(
+        m_webView->GetDevToolsProtocolEventReceiver(L"Target.targetInfoChanged", &receiver));
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                // A target's info like url changed, update it in the target description map.
+                wil::unique_cotaskmem_string jsonMessage;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&jsonMessage));
+                std::wstring targetId = GetJSONStringField(jsonMessage.get(), L"targetId");
+                if (m_devToolsTargetDescriptionMap.find(targetId) !=
+                    m_devToolsTargetDescriptionMap.end())
+                {
+                    // This is a target that we are interested in, update description.
+                    std::wstring type = GetJSONStringField(jsonMessage.get(), L"type");
+                    std::wstring url = GetJSONStringField(jsonMessage.get(), L"url");
+                    m_devToolsTargetDescriptionMap[targetId] = type + L"," + url;
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_targetInfoChangedToken));
+    // Enable Runtime events for the default target of top page to receive Runtime.consoleAPICalled events, which is fired when console.log is called.
+    m_webView->CallDevToolsProtocolMethod(L"Runtime.enable", L"{}", nullptr);
+    // Auto attach to iframe and dedicated worker targets created from the default target of top page.
+    m_webView->CallDevToolsProtocolMethod(
+        L"Target.setAutoAttach",
+        LR"({"autoAttach":true,"waitForDebuggerOnStart":false,"flatten":true})", nullptr);
+}
+```
+## WinRT and .NET    
+```c#
+// The sessions and targets descriptions are tracked by 2 dictionaries:
+// SessionId to TargetId dictionary: m_devToolsSessionMap;
+// TargetId to description dictionary, where description is "<target type>,<target url>": m_devToolsTargetInfoMap
+// GetJSONStringField is a helper function that can retrieve a string field from a json message.
+
+private void CoreWebView2_ConsoleAPICalled(CoreWebView2 sender, CoreWebView2DevToolsProtocolEventReceivedEventArgs args)
+{
+   // Figure out which target the console.log comes from
+   string eventSource;
+   string sessionId = args.SessionId;
```

Comment:
(Just verifying that the C++ samples are version-aware and support downlevel, but the C# samples are not version-aware and require current version.)

Reply:
That is true. All out C# sample code requires current version, and doesn't wrap around calls with try-catch. Will call it out.

Reply:
Yes, previously we decided not to have try/catch for all C# samples.